FastSimulation/Calorimetry/src/CalorimetryManager.cc

```diff
@@ -140,6 +141,61 @@ CalorimetryManager::CalorimetryManager(FSimEvent * aSimEvent,
        fastMuHCAL.getParameter<bool>("MultipleScattering") )
     theMuonHcalEffects = new MaterialEffects(fastMuHCAL);
 
+  auto ecalResponceParameters = fastCalo.getParameter<edm::ParameterSet>("ECALResponceScalingParameters");
+  initECALScaleHistos( ecalResponceParameters );
+}
+
+void CalorimetryManager::initECALScaleHistos( const edm::ParameterSet& pset ) {
+
+  // Get the filename and histogram name
+  doEcalResponseScaling_ = pset.getParameter<bool>("doScalig");
+  if( ! doEcalResponseScaling_ ) return;
+  std::string ecalScalesFileName_ = pset.getUntrackedParameter<std::string>("fileName");
+  std::string ecalScalesHistogramName_ = pset.getUntrackedParameter<std::string>("histogramName");
+
+  // Read histo
+  edm::FileInPath myDataFile( ecalScalesFileName_ ); // TODO: check if this is necessary
+  TFile * myFile = new TFile( myDataFile.fullPath().c_str(), "READ" );
+
+  gROOT->cd(); // create histogram in memory
+  auto obj = myFile->Get( ecalScalesHistogramName_.c_str() );
+
+  // Check if histogram exists in file
+  if(!obj) {
+    throw cms::Exception( "FileReadError" )
+      << "Histogram \"" << ecalScalesHistogramName_
+      << "\" not found in file \"" << ecalScalesFileName_
+      << "\", used for correcting the response of the ECAL in FastSim.\n";
+  }
+  ecalScales_ = new TH3F( *((TH3F*)obj) );
+
+  myFile->Close();
+  delete myFile;
+
+  // By definition, some genE bins are not filled. Since TH3 has no variable Rebin() method,
+  // we create an auxiliary genE histogram with variable binning:
+  auto oldBinH = ecalScales_->Project3D("x");
```

Comment:
not a very transparent way of working,
there are better options,
let's discuss this over skype